dcos_test_utils/gce.py

```diff
@@ -0,0 +1,174 @@
+import os
+import json
+import logging
+from googleapiclient.errors import HttpError
+from googleapiclient import discovery
+from oauth2client.client import GoogleCredentials
+from retrying import retry
+
+log = logging.getLogger(__name__)
+
+
+def catch_http_exceptions(f):
+    def handle_exception(*args, **kwargs):
+        try:
+            return f(*args, **kwargs)
+        except HttpError as e:
+            if 'HttpError 404' in str(e):
```

Comment:
Second this. it looks like this HttpErrors use exactly that https://github.com/google/google-api-python-client/blob/master/googleapiclient/errors.py#L69 And that is a much better way to detect codes